### PR TITLE
Add stylelint to ensure no re-layout high performance css animation

### LINKED ISSUE #3669: Add animation stylelint rule to prevent re-layout animations
**Describe the bug**
Toggle button animation causes re-layout update on animation. Noticed a slight delay on toggle animation from width layout update on animation. Add animation stylelint rule to prevent re-layout animations for [higher performance](https://www.html5rocks.com/en/tutorials/speed/high-performance-animations/).

### PR DESCRIPTION
**Pull request by edplato**
## Description

Fixes #3669  .

- [x] Add stylelint plugin to ensure no re-layout high performance css animation using `stylelint-high-performance-animation`. Paint properties are ignored as they are a lesser performance hit, but that could be activated in the future.
- [x] Update static/conditional styles per lint errors/warnings where possible
- [x] Pad classification list - the active border shadow of the toggles was getting cut off on the left and top (first in list).
- [x] Move `link.css` code to `share.css`

## How To Test

1. Look for browser specific animation style change issues. 

## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview

### CODE REVIEW

File: web/css/palettes.css [hunk 1]
```diff
@@ -135,14 +135,14 @@
 
 .wv-palettes-panel .wv-running-bar::before {
   top: -1px;
-  left: -5px;
+  left: -4px;
```
Comment (jasontk19) [inline web/css/palettes.css:138]: This looks off to me now:
Screen Shot 2021-08-09 at 3 30 18 PM

Here's what it looked like before:
Screen Shot 2021-08-09 at 3 30 47 PM

Comment (jasontk19) [reply web/css/palettes.css:138]: Or perhaps it's actually due to the changes in `web/js/components/sidebar/paletteLegend.js`

**Review by jasontk19 — CHANGES_REQUESTED** [review]
Everything looks good except the running data bar is a little off now.

File: web/css/palettes.css (hunk 1 above)
Comment (edplato) [reply web/css/palettes.css]: I made a few changes there that shifted it over. Just revised that part in latest to fix.

**Review by jasontk19 — APPROVED** [review]

**Pull request merged**